.github/workflows/ci.yml

```diff
@@ -38,7 +38,64 @@ jobs:
 
       - name: Install required packages
         run: |
-          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
 
       - name: Run Linters
         uses: pre-commit/action@v3.0.0
+
+  install-job:
+    name: "Install package"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install model generator package
+        run: |
+          pip install .
+
+  run-tests:
+    name: "Run tests"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install required packages
+        run: |
+          pip install -r requirements-dev.txt
+          pip install -r requirements.txt
+          pip install -e .
+
+      - name: unit test
+        shell: bash
+        run: |
+          pytest \
+            --junit-xml=./results/UnitTest/junit.xml \
+            --cov-report=xml:results/CodeCoverage/cobertura-coverage.xml \
+            --cov \
+            --cov-branch
+
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v3
+        if: always()
+        with:
+          report_paths: ./results/UnitTest/junit.xml
+          summary: true
+          update_check: true
+          annotate_only: true
+
+      - uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: results/CodeCoverage/cobertura-coverage.xml
+          badge: true
+          format: markdown
+          hide_complexity: true
+          indicators: true
+          output: both
```

Comment:
add the following to show the coverage in the job summary
```
 - run: |
          cat code-coverage-results.md >> $GITHUB_STEP_SUMMARY
```

Reply:
done